src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,428 @@
+                                             "Obtain refresh token for registry {}".format(login_server),
+                                             ignore_errors)
+        return accumulated_errors # We can't continue if the refresh token is not received
+
+    print_pass("Obtain refresh token for registry {}".format(login_server))
+
+    from json import loads
+
+    refresh_token = loads(response.content.decode("utf-8"))["refresh_token"]
+    authhost = urlunparse((authurl[0], authurl[1], '/oauth2/token', '', '', ''))
+    content = {
+        'grant_type': 'refresh_token',
+        'service': login_server,
+        'scope': 'registry:catalog:*',
+        'refresh_token': refresh_token
+    }
+
+    response = requests.post(authhost, urlencode(content), headers=headers)
+
+    if response.status_code not in [200]:
+        accumulated_errors |= _error_handler(Errors.CONNECTIVITY_ACCESS_TOKEN_ERROR,
+                                             "Obtain access token for registry {}".format(login_server),
+                                             ignore_errors)
+    else:
+        print_pass("Obtain access token for registry {}".format(login_server))
+
+    return accumulated_errors
+
+
+def _check_health_connectivity(cmd, registry_name, ignore_errors):
+    accumulated_errors = 0
+
+    if registry_name is None:
+        print_warning("Registry name must be provided to check connectivity.")
+        return accumulated_errors
+
+    try:
+        registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+        login_server = registry.login_server
+    except CLIError:
+        from ._docker_utils import get_login_server_suffix
+        suffix = get_login_server_suffix(cmd.cli_ctx)
+        if suffix is None:
+            suffix = '.azurecr.io'
+        login_server = registry_name + suffix
+        print_warning('Could not load login server for {0}, using default {1}.'.format(registry_name, login_server))
+
+    accumulated_errors |= _get_registry_status(login_server, ignore_errors)
+    if accumulated_errors == 0: # Can't continue if DNS lookup failed
+        accumulated_errors |= _get_endpoint_and_token_status(cmd, login_server, ignore_errors)
+
+    return accumulated_errors
+
+
+## Checks for docker pull ##
+
+
+def _pull_docker_image(skip_confirmation, ignore_errors):
+    accumulated_errors = 0
+    image = "mcr.microsoft.com/mcr/hello-world"
+    tag = "latest"
+
+    if not skip_confirmation:
+        from knack.prompting import prompt_y_n
+        confirmation = prompt_y_n("This will pull the image {}:{}. Proceed?".format(image, tag))
+
+    if not confirmation:
+        print_warning("Skipping pull check.")
+        return accumulated_errors
+
+    # If there's no docker command there's no reason to proceed with operation
+    try:
+        docker_command = get_docker_command()
+    except CLIError:
+        accumulated_errors |= _error_handler(Errors.DOCKER_COMMAND_ERROR,
```

Comment:
There could be different errors thrown from `get_docker_command` such as docker is not installed, docker daemon is not running, etc. Why not print the raw error message to provide more context of what went wrong?